plugins/modules/aws_s3_bucket_info.py

```diff
@@ -46,6 +167,10 @@
   sample:
     - creation_date: '2017-07-06 15:05:12 +00:00'
       name: my_bucket
+      # bucket facts if requested
+      bucket_location: dictionary data
+      bucket_cors: dictionary data
+      # ...etc
   type: list
```

Comment:
```suggestion
  sample:
    - creation_date: '2017-07-06 15:05:12 +00:00'
      name: my_bucket
      # bucket facts if requested
      bucket_location: dictionary data
      bucket_cors: dictionary data
      # ...etc
  type: list
```

It would be good to define what you actually expect to be returned here

Reply:
@tremble Thank you very much for review & suggested changes. I've tried to fix all the coding and 'wording'. I'll probably need some info around what you would like to see in documentation / returned data sample.

This is what is returned when you ask for all the facts:
```
"item": {
        "bucket_accelerate_configuration": {},
        "bucket_acl": {
            "Grants": [
                {
                    "Grantee": {
                        "DisplayName": "xxx",
                        "ID": "<UUID>",
                        "Type": "CanonicalUser"
                    },
                    "Permission": "FULL_CONTROL"
                }
            ],
            "Owner": {
                "DisplayName": "user",
                "ID": "<UUID>"
            }
        },
        "bucket_cors": {},
        "bucket_encryption": {},
        "bucket_lifecycle_configuration": {},
        "bucket_location": {
            "LocationConstraint": "us-east-1"
        },
        "bucket_logging": {},
        "bucket_notification_configuration": {},
        "bucket_ownership_controls": {},
        "bucket_policy": {},
        "bucket_policy_status": {},
        "bucket_replication": {},
        "bucket_request_payment": {
            "Payer": "BucketOwner"
        },
        "bucket_tagging": {
            "CamelCase": "SimpleCamelCase",
            "Title Case": "Hello Cruel World",
            "lowercase spaced": "hello cruel world",
            "snake_case": "simple_snake_case"
        },
        "bucket_website": {},
        "creation_date": "2021-01-27T13:08:28+00:00",
        "name": "<bucket_name>",
        "public_access_block": {}
    },
```

Some entries are empty. For example CORS. It will contain your CORS if it's defined. etc.

Do you want me to add something like:
```
sample:
    - creation_date: '2017-07-06 15:05:12 +00:00'
      name: my_bucket
      # bucket properties are returned if requested
      # bucket_facts:
      #     bucket_cors: true #Return bucket CORS configuration
      #
      # each fact returns empty dictionary or corresponding configuration data 
      bucket_cors: {}
      bucket_encryption: {}
      bucket_lifecycle_configuration: {}
      bucket_location: {
            "LocationConstraint": "<bucket region>"
      },
      bucket_logging: {},
      bucket_notification_configuration: {}
      bucket_ownership_controls: {},
      bucket_policy: {}
      bucket_policy_status: {}
      bucket_replication: {}
      bucket_request_payment: {}
      bucket_tagging: {}
      bucket_website: {}
```

And extend `bucket_facts:` DOC section:
```
  bucket_facts:
    description:
      - Retrieve requested S3 bucket detailed information
      - Each bucket_X option executes one API call, hence many options=true will case slower module execution
      - You can limit buckets by using I(name) or I(name_filter) option
>>>
  bucket_facts:
    short_description: 
      - Retrieve requested S3 bucket detailed information
    description:
      - Each bucket_X suboption executes one API call, hence many options=true will case slower module execution
      - You can also limit buckets by using I(name) or I(name_filter) option
      - ...
```

Reply:
Doc & return updated as requested - https://github.com/ansible-collections/community.aws/pull/260/commits/dec9b0522ae4921b0bb443cd654f62697df8aaca